daemons/fenced/fenced_commands.c

```diff
@@ -3275,22 +3264,18 @@ handle_fence_request(pcmk__request_t *request)
                 return fenced_construct_reply(request->xml, NULL,
                                               &request->result);
             }
+            alternate_host = check_alternate_host(target);
 
```

Comment:
as this is the only branch that might create a relay-request the logic might
be even clearer if we pull the relay-request creation in here instead of 
checking later on. Without further refactoring that block would then need
to be do {} while(0) - style.

Reply:
yeah this whole function could use some refactoring